ethereum/consensus-core/src/consensus_core.rs

```diff
@@ -307,6 +307,8 @@ pub fn verify_generic_update<S: ConsensusSpec>(
         } else {
             return Err(ConsensusError::InvalidFinalityProof.into());
         }
+    } else {
+        return Err(ConsensusError::MissingFinalizedHeader.into());
```

Comment:
It is possible to have an update that does not contain a finalized header (see optimisitc updates).

Reply:
Ah, right